src/utils.py

```diff
@@ -1857,6 +1857,40 @@ def sample_subsets(universe: Sequence[_T], num_samples: int, min_set_size: int,
         yield sample
 
 
+def create_dataset_filename_str(
+        saving_ground_atoms: bool,
+        online_learning_cycle: Optional[int] = None) -> Tuple[str, str]:
+    """Generate strings to be used for the filename for a dataset file that is
+    about to be saved.
+
+    Returns a tuple of strings where the first element is the dataset
+    filename itself and the second is a template string used to generate
+    it. If saving_ground_atoms is True, then we will name the file with
+    a "_ground_atoms" suffix.
+    """
+    # Setup the dataset filename for saving/loading GroundAtoms.
+    regex = r"(\d+)"
+    suffix_str = ""
+    if online_learning_cycle is not None:
```

Comment:
the convention we've used thus far is to include online_learning_cycle regardless (so None appears in the str sometimes). this is helpful when parsing filenames, so we can assume the number of "__" separators is always the same.